src/features/game/events/landExpansion/enhancedRestock.ts

```diff
@@ -0,0 +1,76 @@
+import Decimal from "decimal.js-light";
+import { INITIAL_STOCK } from "features/game/lib/constants";
+import { getKeys } from "features/game/types/decorations";
+import { GameState } from "features/game/types/game";
+import { SEEDS } from "features/game/types/seeds";
+import { TREASURE_TOOLS, WORKBENCH_TOOLS } from "features/game/types/tools";
+import { produce } from "immer";
+import { NPCName } from "lib/npcs";
+import { onboardingAnalytics } from "lib/onboardingAnalytics";
+
+export type RestockNPC = Extract<NPCName, "betty" | "jafar" | "blacksmith">;
+
+export type EnhancedRestockAction = {
+  type: "shops.restocked.enhanced";
+  npc: RestockNPC;
+};
+
+type Options = {
+  state: Readonly<GameState>;
+  action: EnhancedRestockAction;
+};
+
+type RestockObject = {
+  restockItem: object;
+  gemPrice: number;
+  shopName: "market" | "workbench" | "treasure shop";
+};
+
+export const RestockItems: Record<RestockNPC, RestockObject> = {
```

Comment:
In aggregate these are more expensive than the current single restock.

Further, betty + blacksmith are also more expensive than today.

So, just on that alone it's a downgrade. :|

Perhaps some tuning to make this more appealing? @adamhannigan